test/framework/src/main/java/org/elasticsearch/search/aggregations/BaseAggregationTestCase.java

```diff
@@ -61,10 +61,25 @@ public void testFromXContent() throws IOException {
     public void testSupportsConcurrentExecution() {
         AB builder = createTestAggregatorBuilder();
         boolean supportsConcurrency = builder.supportsConcurrentExecution();
+        if (supportsConcurrency == false) {
+            assertFalse(builder.supportsOffloadingSequentialCollection());
```

Comment:
Would this always be correct? Terms aggregation doesn't support concurrency now, but should support offloading?

Reply:
I think we always assert that if an agg does support concurrency then it also supports offloading?

Reply:
yes, I got this wrong. thanks for raising this.